projects/plugins/jetpack/class.jetpack.php

```diff
@@ -3468,6 +3468,17 @@ function admin_init() {
 		add_filter( 'manage_users_columns', array( $this, 'jetpack_icon_user_connected' ) );
 		add_action( 'manage_users_custom_column', array( $this, 'jetpack_show_user_connected_icon' ), 10, 3 );
 		add_action( 'admin_print_styles', array( $this, 'jetpack_user_col_style' ) );
+
+		// Accept and store a partner coupon if present, and redirect to Jetpack connection screen
+		if ( isset( $_GET['jetpack-partner-coupon'] ) ) {
+			update_option( 'jetpack_partner_coupon', sanitize_text_field( $_GET['jetpack-partner-coupon'] ) );
```

Comment:
I think this would be the right place to add a "date added" option to auto-purge the coupon field (after a month?) 😄